palette/src/rgb/packed.rs

```diff
@@ -0,0 +1,364 @@
+use core::marker::PhantomData;
+
+use crate::rgb::{Rgb, RgbStandard, Rgba};
+use crate::{Alpha, Pixel};
+
+/// RGBA color packed into a 32-bit unsigned integer, defaults to ARGB ordering.
+///
+/// Packed integer type represented in `u32`. Two hexadecimal digits (8-bits)
+/// express each value of the Red, Green, Blue, and Alpha components in the
+/// RGBA color.
+///
+/// Note that conversion from float to integer component types in `palette` maps
+/// the floating point value to the integer's value range and then rounds the
+/// result before casting to the integer type. An example of the difference in
+/// conversion methods is shown below.
+///
+/// ```
+/// use approx::assert_relative_eq;
+/// use palette::{PackedRgb, Srgb, Srgba};
+/// use palette::rgb::{Argb, RgbaPacked};
+///
+/// let packed: PackedRgb = Srgb::new(0.5, 0.0, 0.5).into_format().into();
+/// assert_eq!(0x0080_0080, packed.color);
+///
+/// let unpacked: Srgba<u8> = PackedRgb::<RgbaPacked>::from(0x80FF_FF80).into();
+/// assert_relative_eq!(
+///     Srgba::new(0.5, 1.0, 1.0, 0.5),
+///     unpacked.into_format(),
+///     epsilon = 0.01
+/// );
+///
+/// // By default, `PackedRGB` uses `Argb` order to make entering 6-digit hex
+/// // numbers more convenient
+/// let unpacked = Srgb::<u8>::from_hex::<Argb>(0xFF8000);
+/// assert_eq!(Srgb::<u8>::new(0xFF, 0x80, 0x00), unpacked);
+///
+/// // The second assert is essentially how library components are converted
+/// let float = 0.5f32 * 255.0;
+/// assert_eq!(0x7F, float as u32);
+/// assert_eq!(0x80, float.round() as u32);
+/// ```
+///
+/// When an `Srgb` type is packed, the alpha value will be `0x00` in the
+/// corresponding `u32`. Converting from a packed color type back to an `Srgb`
+/// type will disregard the alpha value.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Pixel)]
+#[palette_internal]
+#[repr(C)]
+pub struct PackedRgb<C: RgbChannels = Argb> {
```

Comment:
The `Rgb` suffix may be redundant, given by the context and the type parameter. I'm definitely open for other names, both here and elsewhere.

Reply:
I can't think of another name, I think `Packed` works best for now. No other names stick out as bad to me right now.